scripts/make_tor_targets.py

```diff
@@ -0,0 +1,114 @@
+import json
+import hashlib
+
+from urllib.request import urlopen
+
+DIRAUTH_URL = "https://gitweb.torproject.org/tor.git/plain/src/app/config/auth_dirs.inc"
+BRIDGES_URL = "https://bridges.torproject.org/moat/circumvention/builtin"
+
+def parse_params(parts : list[str]) -> dict[str, str]:
+    params = {}
+    for p in parts:
+        k, v = p.split("=")
+        params[k] = [v]
+    return params
+
+def parse_bridge_line(line : str) -> tuple[str, dict]:
+    bridge = {}
+    parts = line.split(" ")
+    bridge["protocol"] = parts[0]
+    bridge["address"] = parts[1]
+    bridge["fingerprint"] = parts[2]
+    bridge["params"] = parse_params(parts[3:])
+
+    bridge_id = hashlib.sha256(
+            bridge["address"].encode("ascii") + bridge["fingerprint"].encode("utf-8")
+    ).hexdigest()
+
+    return bridge_id, bridge
+
+def get_bridges():
+    with urlopen(BRIDGES_URL) as resp:
+        j = json.loads(resp.read())
+
+    bridges = {}
+    for b in j["obfs4"]:
+        bridge_id, bd = parse_bridge_line(b)
+        assert bridge_id not in bridges
+        bridges[bridge_id] = bd
+    return bridges
+
+def parse_dirauth(line : str) -> dict:
+    da = {}
+    parts = line.split(" ")
+    da["name"] = parts[0]
+    assert parts[1].startswith("orport=")
+    da["or_port"] = parts[1].lstrip("orport=")
+    da["dir_address"] = parts[-11]
+    da["fingerprint"] = "".join(parts[-10:])
+    return da
+
+def get_dirauths():
+    with urlopen(DIRAUTH_URL) as resp:
+        config = resp.read().decode("utf-8")
+
+    dir_auths = {}
+
+    da_lines = []
+    current_line = ""
+    is_done = False
+    for line in config.split("\n"):
+        if is_done is True:
+            current_line = ""
+            is_done = False
+
+        if line.endswith(","):
+            is_done = True
+            line = line.rstrip(",")
+
+        line = line.strip().lstrip('"').rstrip('"')
+        current_line += line
+        if is_done:
+            da_lines.append(current_line)
+
+    for line in da_lines:
+        da = parse_dirauth(line)
+
+        or_address = da["dir_address"].split(":")[0] + ":" + da["or_port"]
```

Comment:
```suggestion
        # Note: this algorithm only manipulates IPv4 addresses
        or_address = da["dir_address"].split(":")[0] + ":" + da["or_port"]
```

Honestly here I would be happier to use correct parsing of endpoints rather than splitting by `:` because that would break when using IPv6. I would recommend borrowing more robust code to perform this kind of splitting from here: https://github.com/bassosimone/websteps-illustrated/blob/ef7a88b15ac5d13fb95525551e9cb078853cfa7b/python/websteps.py#L325